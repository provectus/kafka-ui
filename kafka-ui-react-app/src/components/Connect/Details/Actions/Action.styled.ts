import styled from 'styled-components';

export const ConnectorActionsWrapperStyled = styled.div`
  display: flex;
  flex-wrap: wrap;
  align-items: center;
  gap: 8px;
`;
export const ButtonLabel = styled.span`
  margin-right: 11.5px;
`;
export const RestartButton = styled.div`
  padding: 0 12px;
  border: none;
  border-radius: 4px;
  display: flex;
  -webkit-align-items: center;
  background: ${({ theme }) => theme.button.primary.backgroundColor.normal};
  color: ${({ theme }) => theme.button.primary.color.normal};
  font-size: 14px;
  font-weight: 500;
  height: 32px;
`;
