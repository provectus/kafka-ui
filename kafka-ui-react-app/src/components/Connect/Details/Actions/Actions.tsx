import React from 'react';
import { useNavigate } from 'react-router-dom';
import { useIsMutating } from '@tanstack/react-query';
import {
  Action,
  ConnectorAction,
  ConnectorState,
  ResourceType,
} from 'generated-sources';
import useAppParams from 'lib/hooks/useAppParams';
import {
  useConnector,
  useDeleteConnector,
  useUpdateConnectorState,
} from 'lib/hooks/api/kafkaConnect';
import {
  clusterConnectorsPath,
  RouterParamsClusterConnectConnector,
} from 'lib/paths';
import { useConfirm } from 'lib/hooks/useConfirm';
import { Dropdown } from 'components/common/Dropdown';
import { ActionDropdownItem } from 'components/common/ActionComponent';
import ChevronDownIcon from 'components/common/Icons/ChevronDownIcon';

import * as S from './Action.styled';

const Actions: React.FC = () => {
  const navigate = useNavigate();
  const routerProps = useAppParams<RouterParamsClusterConnectConnector>();
  const mutationsNumber = useIsMutating();
  const isMutating = mutationsNumber > 0;

  const { data: connector } = useConnector(routerProps);
  const confirm = useConfirm();

  const deleteConnectorMutation = useDeleteConnector(routerProps);
  const deleteConnectorHandler = () =>
    confirm(
      <>
        Are you sure you want to remove <b>{routerProps.connectorName}</b>{' '}
        connector?
      </>,
      async () => {
        try {
          await deleteConnectorMutation.mutateAsync();
          navigate(clusterConnectorsPath(routerProps.clusterName));
        } catch {
          // do not redirect
        }
      }
    );

  const stateMutation = useUpdateConnectorState(routerProps);
  const restartConnectorHandler = () =>
    stateMutation.mutateAsync(ConnectorAction.RESTART);
  const restartAllTasksHandler = () =>
    stateMutation.mutateAsync(ConnectorAction.RESTART_ALL_TASKS);
  const restartFailedTasksHandler = () =>
    stateMutation.mutateAsync(ConnectorAction.RESTART_FAILED_TASKS);
  const pauseConnectorHandler = () =>
    stateMutation.mutateAsync(ConnectorAction.PAUSE);
  const resumeConnectorHandler = () =>
    stateMutation.mutateAsync(ConnectorAction.RESUME);
  return (
    <S.ConnectorActionsWrapperStyled>
      <Dropdown
        label={
          <S.RestartButton>
            <S.ButtonLabel>Restart</S.ButtonLabel>
            <ChevronDownIcon />
          </S.RestartButton>
        }
      >
        {connector?.status.state === ConnectorState.RUNNING && (
          <ActionDropdownItem
            onClick={pauseConnectorHandler}
            disabled={isMutating}
            permission={{
              resource: ResourceType.CONNECT,
              action: Action.EDIT,
              value: routerProps.connectorName,
            }}
          >
            Pause
          </ActionDropdownItem>
        )}
        {connector?.status.state === ConnectorState.PAUSED && (
          <ActionDropdownItem
            onClick={resumeConnectorHandler}
            disabled={isMutating}
            permission={{
              resource: ResourceType.CONNECT,
              action: Action.EDIT,
              value: routerProps.connectorName,
            }}
          >
            Resume
          </ActionDropdownItem>
        )}
        <ActionDropdownItem
          onClick={restartConnectorHandler}
          disabled={isMutating}
          permission={{
            resource: ResourceType.CONNECT,
            action: Action.EDIT,
            value: routerProps.connectorName,
          }}
        >
          Restart Connector
        </ActionDropdownItem>
        <ActionDropdownItem
          onClick={restartAllTasksHandler}
          disabled={isMutating}
          permission={{
            resource: ResourceType.CONNECT,
            action: Action.EDIT,
            value: routerProps.connectorName,
          }}
        >
          Restart All Tasks
        </ActionDropdownItem>
        <ActionDropdownItem
          onClick={restartFailedTasksHandler}
          disabled={isMutating}
          permission={{
            resource: ResourceType.CONNECT,
            action: Action.EDIT,
            value: routerProps.connectorName,
          }}
        >
          Restart Failed Tasks
        </ActionDropdownItem>
      </Dropdown>
      <Dropdown>
        <ActionDropdownItem
          onClick={deleteConnectorHandler}
          disabled={isMutating}
          danger
          permission={{
            resource: ResourceType.CONNECT,
            action: Action.DELETE,
            value: routerProps.connectorName,
          }}
        >
          Delete
        </ActionDropdownItem>
      </Dropdown>
    </S.ConnectorActionsWrapperStyled>
  );
};

export default Actions;
