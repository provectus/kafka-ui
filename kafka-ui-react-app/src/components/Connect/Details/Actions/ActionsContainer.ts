import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import { RootState } from 'redux/interfaces';
import {
  deleteConnector,
  restartConnector,
  restartTasks,
  pauseConnector,
  resumeConnector,
} from 'redux/reducers/connect/connectSlice';
import {
  getIsConnectorDeleting,
  getConnectorStatus,
  getIsConnectorActionRunning,
} from 'redux/reducers/connect/selectors';

import Actions from './Actions';

const mapStateToProps = (state: RootState) => ({
  isConnectorDeleting: getIsConnectorDeleting(state),
  connectorStatus: getConnectorStatus(state),
  isConnectorActionRunning: getIsConnectorActionRunning(state),
});

const mapDispatchToProps = {
  deleteConnector,
  restartConnector,
  restartTasks,
  pauseConnector,
  resumeConnector,
};

export default withRouter(
  connect(mapStateToProps, mapDispatchToProps)(Actions)
);
