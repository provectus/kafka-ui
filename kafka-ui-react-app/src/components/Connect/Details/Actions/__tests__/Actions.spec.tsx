import React from 'react';
import { render, WithRoute } from 'lib/testHelpers';
import { clusterConnectConnectorPath } from 'lib/paths';
import Actions from 'components/Connect/Details/Actions/Actions';
import { ConnectorAction, ConnectorState } from 'generated-sources';
import { screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import {
  useConnector,
  useUpdateConnectorState,
} from 'lib/hooks/api/kafkaConnect';
import { connector } from 'lib/fixtures/kafkaConnect';
import set from 'lodash/set';

const mockHistoryPush = jest.fn();
const deleteConnector = jest.fn();
const cancelMock = jest.fn();

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockHistoryPush,
}));

jest.mock('lib/hooks/api/kafkaConnect', () => ({
  useConnector: jest.fn(),
  useDeleteConnector: jest.fn(),
  useUpdateConnectorState: jest.fn(),
}));

const expectActionButtonsExists = () => {
  expect(screen.getByText('Restart Connector')).toBeInTheDocument();
  expect(screen.getByText('Restart All Tasks')).toBeInTheDocument();
  expect(screen.getByText('Restart Failed Tasks')).toBeInTheDocument();
  expect(screen.getByText('Delete')).toBeInTheDocument();
};
const afterClickDropDownButton = async () => {
  const dropDownButton = screen.getAllByRole('button');
  await userEvent.click(dropDownButton[1]);
};
const afterClickRestartButton = async () => {
  const dropDownButton = screen.getByText('Restart');
  await userEvent.click(dropDownButton);
};
describe('Actions', () => {
  afterEach(() => {
    mockHistoryPush.mockClear();
    deleteConnector.mockClear();
    cancelMock.mockClear();
  });

  describe('view', () => {
    const route = clusterConnectConnectorPath();
    const path = clusterConnectConnectorPath(
      'myCluster',
      'myConnect',
      'myConnector'
    );

    const renderComponent = () =>
      render(
        <WithRoute path={route}>
          <Actions />
        </WithRoute>,
        { initialEntries: [path] }
      );

    it('renders buttons when paused', async () => {
      (useConnector as jest.Mock).mockImplementation(() => ({
        data: set({ ...connector }, 'status.state', ConnectorState.PAUSED),
      }));
      renderComponent();
      await afterClickRestartButton();
      expect(screen.getAllByRole('menuitem').length).toEqual(4);
      expect(screen.getByText('Resume')).toBeInTheDocument();
      expect(screen.queryByText('Pause')).not.toBeInTheDocument();
      expectActionButtonsExists();
    });

    it('renders buttons when failed', async () => {
      (useConnector as jest.Mock).mockImplementation(() => ({
        data: set({ ...connector }, 'status.state', ConnectorState.FAILED),
      }));
      renderComponent();
      await afterClickRestartButton();
      expect(screen.getAllByRole('menuitem').length).toEqual(3);
      expect(screen.queryByText('Resume')).not.toBeInTheDocument();
      expect(screen.queryByText('Pause')).not.toBeInTheDocument();
      expectActionButtonsExists();
    });

    it('renders buttons when unassigned', async () => {
      (useConnector as jest.Mock).mockImplementation(() => ({
        data: set({ ...connector }, 'status.state', ConnectorState.UNASSIGNED),
      }));
      renderComponent();
      await afterClickRestartButton();
      expect(screen.getAllByRole('menuitem').length).toEqual(3);
      expect(screen.queryByText('Resume')).not.toBeInTheDocument();
      expect(screen.queryByText('Pause')).not.toBeInTheDocument();
      expectActionButtonsExists();
    });

    it('renders buttons when running connector action', async () => {
      (useConnector as jest.Mock).mockImplementation(() => ({
        data: set({ ...connector }, 'status.state', ConnectorState.RUNNING),
      }));
      renderComponent();
      await afterClickRestartButton();
      expect(screen.getAllByRole('menuitem').length).toEqual(4);
      expect(screen.queryByText('Resume')).not.toBeInTheDocument();
      expect(screen.getByText('Pause')).toBeInTheDocument();
      expectActionButtonsExists();
    });

    describe('mutations', () => {
      beforeEach(() => {
        (useConnector as jest.Mock).mockImplementation(() => ({
          data: set({ ...connector }, 'status.state', ConnectorState.RUNNING),
        }));
      });

      it('opens confirmation modal when delete button clicked', async () => {
        renderComponent();
        await afterClickDropDownButton();
        await waitFor(async () =>
          userEvent.click(screen.getByRole('menuitem', { name: 'Delete' }))
        );
        expect(screen.getByRole('dialog')).toBeInTheDocument();
      });

      it('calls restartConnector when restart button clicked', async () => {
        const restartConnector = jest.fn();
        (useUpdateConnectorState as jest.Mock).mockImplementation(() => ({
          mutateAsync: restartConnector,
        }));
        renderComponent();
        await afterClickRestartButton();
        await userEvent.click(
          screen.getByRole('menuitem', { name: 'Restart Connector' })
        );
        expect(restartConnector).toHaveBeenCalledWith(ConnectorAction.RESTART);
      });

      it('calls restartAllTasks', async () => {
        const restartAllTasks = jest.fn();
        (useUpdateConnectorState as jest.Mock).mockImplementation(() => ({
          mutateAsync: restartAllTasks,
        }));
        renderComponent();
        await afterClickRestartButton();
        await userEvent.click(
          screen.getByRole('menuitem', { name: 'Restart All Tasks' })
        );
        expect(restartAllTasks).toHaveBeenCalledWith(
          ConnectorAction.RESTART_ALL_TASKS
        );
      });

      it('calls restartFailedTasks', async () => {
        const restartFailedTasks = jest.fn();
        (useUpdateConnectorState as jest.Mock).mockImplementation(() => ({
          mutateAsync: restartFailedTasks,
        }));
        renderComponent();
        await afterClickRestartButton();
        await userEvent.click(
          screen.getByRole('menuitem', { name: 'Restart Failed Tasks' })
        );
        expect(restartFailedTasks).toHaveBeenCalledWith(
          ConnectorAction.RESTART_FAILED_TASKS
        );
      });

      it('calls pauseConnector when pause button clicked', async () => {
        const pauseConnector = jest.fn();
        (useUpdateConnectorState as jest.Mock).mockImplementation(() => ({
          mutateAsync: pauseConnector,
        }));
        renderComponent();
        await afterClickRestartButton();
        await userEvent.click(screen.getByRole('menuitem', { name: 'Pause' }));
        expect(pauseConnector).toHaveBeenCalledWith(ConnectorAction.PAUSE);
      });

      it('calls resumeConnector when resume button clicked', async () => {
        const resumeConnector = jest.fn();
        (useConnector as jest.Mock).mockImplementation(() => ({
          data: set({ ...connector }, 'status.state', ConnectorState.PAUSED),
        }));
        (useUpdateConnectorState as jest.Mock).mockImplementation(() => ({
          mutateAsync: resumeConnector,
        }));
        renderComponent();
        await afterClickRestartButton();
        await userEvent.click(screen.getByRole('menuitem', { name: 'Resume' }));
        expect(resumeConnector).toHaveBeenCalledWith(ConnectorAction.RESUME);
      });
    });
  });
});
