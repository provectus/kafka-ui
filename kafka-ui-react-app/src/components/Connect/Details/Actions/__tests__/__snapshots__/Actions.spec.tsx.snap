// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Actions view matches snapshot 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0px 12px;
  border: none;
  border-radius: 4px;
  white-space: nowrap;
  background: #4F4FFF;
  color: #FFFFFF;
  font-size: 14px;
  height: 32px;
}

.c1:hover:enabled {
  background: #1717CF;
  color: #FFFFFF;
  cursor: pointer;
}

.c1:active:enabled {
  background: #1414B8;
  color: #FFFFFF;
}

.c1:disabled {
  opacity: 0.5;
  cursor: not-allowed;
}

.c1 a {
  color: white;
}

.c1 i {
  margin-right: 7px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0px 12px;
  border: none;
  border-radius: 4px;
  white-space: nowrap;
  background: #F1F2F3;
  color: #171A1C;
  font-size: 14px;
  height: 32px;
}

.c2:hover:enabled {
  background: #E3E6E8;
  color: #171A1C;
  cursor: pointer;
}

.c2:active:enabled {
  background: #D5DADD;
  color: #171A1C;
}

.c2:disabled {
  opacity: 0.5;
  cursor: not-allowed;
}

.c2 a {
  color: white;
}

.c2 i {
  margin-right: 7px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 8px;
}

<div
  className="c0"
>
  <button
    className="c1"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    <span>
      <i
        className="fas fa-pause"
      />
    </span>
    <span>
      Pause
    </span>
  </button>
  <button
    className="c1"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    <span>
      <i
        className="fas fa-sync-alt"
      />
    </span>
    <span>
      Restart All Tasks
    </span>
  </button>
  <a
    href="/ui/clusters/my-cluster/connects/my-connect/connectors/my-connector/edit"
    onClick={[Function]}
  >
    <button
      className="c1"
      disabled={false}
      type="button"
    >
      <span>
        <i
          className="fas fa-pencil-alt"
        />
      </span>
      <span>
        Edit Config
      </span>
    </button>
  </a>
  <button
    className="c2"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    <span>
      <i
        className="far fa-trash-alt"
      />
    </span>
    <span>
      Delete
    </span>
  </button>
  <mock-ConfirmationModal
    isConfirming={false}
    isOpen={false}
    onCancel={[Function]}
    onConfirm={[Function]}
  >
    Are you sure you want to remove 
    <b>
      my-connector
    </b>
     connector?
  </mock-ConfirmationModal>
</div>
`;

exports[`Actions view matches snapshot when deleting connector 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0px 12px;
  border: none;
  border-radius: 4px;
  white-space: nowrap;
  background: #4F4FFF;
  color: #FFFFFF;
  font-size: 14px;
  height: 32px;
}

.c1:hover:enabled {
  background: #1717CF;
  color: #FFFFFF;
  cursor: pointer;
}

.c1:active:enabled {
  background: #1414B8;
  color: #FFFFFF;
}

.c1:disabled {
  opacity: 0.5;
  cursor: not-allowed;
}

.c1 a {
  color: white;
}

.c1 i {
  margin-right: 7px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0px 12px;
  border: none;
  border-radius: 4px;
  white-space: nowrap;
  background: #F1F2F3;
  color: #171A1C;
  font-size: 14px;
  height: 32px;
}

.c2:hover:enabled {
  background: #E3E6E8;
  color: #171A1C;
  cursor: pointer;
}

.c2:active:enabled {
  background: #D5DADD;
  color: #171A1C;
}

.c2:disabled {
  opacity: 0.5;
  cursor: not-allowed;
}

.c2 a {
  color: white;
}

.c2 i {
  margin-right: 7px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 8px;
}

<div
  className="c0"
>
  <button
    className="c1"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    <span>
      <i
        className="fas fa-pause"
      />
    </span>
    <span>
      Pause
    </span>
  </button>
  <button
    className="c1"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    <span>
      <i
        className="fas fa-sync-alt"
      />
    </span>
    <span>
      Restart All Tasks
    </span>
  </button>
  <a
    href="/ui/clusters/my-cluster/connects/my-connect/connectors/my-connector/edit"
    onClick={[Function]}
  >
    <button
      className="c1"
      disabled={false}
      type="button"
    >
      <span>
        <i
          className="fas fa-pencil-alt"
        />
      </span>
      <span>
        Edit Config
      </span>
    </button>
  </a>
  <button
    className="c2"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    <span>
      <i
        className="far fa-trash-alt"
      />
    </span>
    <span>
      Delete
    </span>
  </button>
  <mock-ConfirmationModal
    isConfirming={true}
    isOpen={false}
    onCancel={[Function]}
    onConfirm={[Function]}
  >
    Are you sure you want to remove 
    <b>
      my-connector
    </b>
     connector?
  </mock-ConfirmationModal>
</div>
`;

exports[`Actions view matches snapshot when failed 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0px 12px;
  border: none;
  border-radius: 4px;
  white-space: nowrap;
  background: #4F4FFF;
  color: #FFFFFF;
  font-size: 14px;
  height: 32px;
}

.c1:hover:enabled {
  background: #1717CF;
  color: #FFFFFF;
  cursor: pointer;
}

.c1:active:enabled {
  background: #1414B8;
  color: #FFFFFF;
}

.c1:disabled {
  opacity: 0.5;
  cursor: not-allowed;
}

.c1 a {
  color: white;
}

.c1 i {
  margin-right: 7px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0px 12px;
  border: none;
  border-radius: 4px;
  white-space: nowrap;
  background: #F1F2F3;
  color: #171A1C;
  font-size: 14px;
  height: 32px;
}

.c2:hover:enabled {
  background: #E3E6E8;
  color: #171A1C;
  cursor: pointer;
}

.c2:active:enabled {
  background: #D5DADD;
  color: #171A1C;
}

.c2:disabled {
  opacity: 0.5;
  cursor: not-allowed;
}

.c2 a {
  color: white;
}

.c2 i {
  margin-right: 7px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 8px;
}

<div
  className="c0"
>
  <button
    className="c1"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    <span>
      <i
        className="fas fa-sync-alt"
      />
    </span>
    <span>
      Restart All Tasks
    </span>
  </button>
  <a
    href="/ui/clusters/my-cluster/connects/my-connect/connectors/my-connector/edit"
    onClick={[Function]}
  >
    <button
      className="c1"
      disabled={false}
      type="button"
    >
      <span>
        <i
          className="fas fa-pencil-alt"
        />
      </span>
      <span>
        Edit Config
      </span>
    </button>
  </a>
  <button
    className="c2"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    <span>
      <i
        className="far fa-trash-alt"
      />
    </span>
    <span>
      Delete
    </span>
  </button>
  <mock-ConfirmationModal
    isConfirming={false}
    isOpen={false}
    onCancel={[Function]}
    onConfirm={[Function]}
  >
    Are you sure you want to remove 
    <b>
      my-connector
    </b>
     connector?
  </mock-ConfirmationModal>
</div>
`;

exports[`Actions view matches snapshot when paused 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0px 12px;
  border: none;
  border-radius: 4px;
  white-space: nowrap;
  background: #4F4FFF;
  color: #FFFFFF;
  font-size: 14px;
  height: 32px;
}

.c1:hover:enabled {
  background: #1717CF;
  color: #FFFFFF;
  cursor: pointer;
}

.c1:active:enabled {
  background: #1414B8;
  color: #FFFFFF;
}

.c1:disabled {
  opacity: 0.5;
  cursor: not-allowed;
}

.c1 a {
  color: white;
}

.c1 i {
  margin-right: 7px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0px 12px;
  border: none;
  border-radius: 4px;
  white-space: nowrap;
  background: #F1F2F3;
  color: #171A1C;
  font-size: 14px;
  height: 32px;
}

.c2:hover:enabled {
  background: #E3E6E8;
  color: #171A1C;
  cursor: pointer;
}

.c2:active:enabled {
  background: #D5DADD;
  color: #171A1C;
}

.c2:disabled {
  opacity: 0.5;
  cursor: not-allowed;
}

.c2 a {
  color: white;
}

.c2 i {
  margin-right: 7px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 8px;
}

<div
  className="c0"
>
  <button
    className="c1"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    <span>
      <i
        className="fas fa-play"
      />
    </span>
    <span>
      Resume
    </span>
  </button>
  <button
    className="c1"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    <span>
      <i
        className="fas fa-sync-alt"
      />
    </span>
    <span>
      Restart All Tasks
    </span>
  </button>
  <a
    href="/ui/clusters/my-cluster/connects/my-connect/connectors/my-connector/edit"
    onClick={[Function]}
  >
    <button
      className="c1"
      disabled={false}
      type="button"
    >
      <span>
        <i
          className="fas fa-pencil-alt"
        />
      </span>
      <span>
        Edit Config
      </span>
    </button>
  </a>
  <button
    className="c2"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    <span>
      <i
        className="far fa-trash-alt"
      />
    </span>
    <span>
      Delete
    </span>
  </button>
  <mock-ConfirmationModal
    isConfirming={false}
    isOpen={false}
    onCancel={[Function]}
    onConfirm={[Function]}
  >
    Are you sure you want to remove 
    <b>
      my-connector
    </b>
     connector?
  </mock-ConfirmationModal>
</div>
`;

exports[`Actions view matches snapshot when running connector action 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0px 12px;
  border: none;
  border-radius: 4px;
  white-space: nowrap;
  background: #4F4FFF;
  color: #FFFFFF;
  font-size: 14px;
  height: 32px;
}

.c1:hover:enabled {
  background: #1717CF;
  color: #FFFFFF;
  cursor: pointer;
}

.c1:active:enabled {
  background: #1414B8;
  color: #FFFFFF;
}

.c1:disabled {
  opacity: 0.5;
  cursor: not-allowed;
}

.c1 a {
  color: white;
}

.c1 i {
  margin-right: 7px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0px 12px;
  border: none;
  border-radius: 4px;
  white-space: nowrap;
  background: #F1F2F3;
  color: #171A1C;
  font-size: 14px;
  height: 32px;
}

.c2:hover:enabled {
  background: #E3E6E8;
  color: #171A1C;
  cursor: pointer;
}

.c2:active:enabled {
  background: #D5DADD;
  color: #171A1C;
}

.c2:disabled {
  opacity: 0.5;
  cursor: not-allowed;
}

.c2 a {
  color: white;
}

.c2 i {
  margin-right: 7px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 8px;
}

<div
  className="c0"
>
  <button
    className="c1"
    disabled={true}
    onClick={[Function]}
    type="button"
  >
    <span>
      <i
        className="fas fa-pause"
      />
    </span>
    <span>
      Pause
    </span>
  </button>
  <button
    className="c1"
    disabled={true}
    onClick={[Function]}
    type="button"
  >
    <span>
      <i
        className="fas fa-sync-alt"
      />
    </span>
    <span>
      Restart All Tasks
    </span>
  </button>
  <a
    href="/ui/clusters/my-cluster/connects/my-connect/connectors/my-connector/edit"
    onClick={[Function]}
  >
    <button
      className="c1"
      disabled={true}
      type="button"
    >
      <span>
        <i
          className="fas fa-pencil-alt"
        />
      </span>
      <span>
        Edit Config
      </span>
    </button>
  </a>
  <button
    className="c2"
    disabled={true}
    onClick={[Function]}
    type="button"
  >
    <span>
      <i
        className="far fa-trash-alt"
      />
    </span>
    <span>
      Delete
    </span>
  </button>
  <mock-ConfirmationModal
    isConfirming={false}
    isOpen={false}
    onCancel={[Function]}
    onConfirm={[Function]}
  >
    Are you sure you want to remove 
    <b>
      my-connector
    </b>
     connector?
  </mock-ConfirmationModal>
</div>
`;

exports[`Actions view matches snapshot when unassigned 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0px 12px;
  border: none;
  border-radius: 4px;
  white-space: nowrap;
  background: #4F4FFF;
  color: #FFFFFF;
  font-size: 14px;
  height: 32px;
}

.c1:hover:enabled {
  background: #1717CF;
  color: #FFFFFF;
  cursor: pointer;
}

.c1:active:enabled {
  background: #1414B8;
  color: #FFFFFF;
}

.c1:disabled {
  opacity: 0.5;
  cursor: not-allowed;
}

.c1 a {
  color: white;
}

.c1 i {
  margin-right: 7px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0px 12px;
  border: none;
  border-radius: 4px;
  white-space: nowrap;
  background: #F1F2F3;
  color: #171A1C;
  font-size: 14px;
  height: 32px;
}

.c2:hover:enabled {
  background: #E3E6E8;
  color: #171A1C;
  cursor: pointer;
}

.c2:active:enabled {
  background: #D5DADD;
  color: #171A1C;
}

.c2:disabled {
  opacity: 0.5;
  cursor: not-allowed;
}

.c2 a {
  color: white;
}

.c2 i {
  margin-right: 7px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 8px;
}

<div
  className="c0"
>
  <button
    className="c1"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    <span>
      <i
        className="fas fa-sync-alt"
      />
    </span>
    <span>
      Restart All Tasks
    </span>
  </button>
  <a
    href="/ui/clusters/my-cluster/connects/my-connect/connectors/my-connector/edit"
    onClick={[Function]}
  >
    <button
      className="c1"
      disabled={false}
      type="button"
    >
      <span>
        <i
          className="fas fa-pencil-alt"
        />
      </span>
      <span>
        Edit Config
      </span>
    </button>
  </a>
  <button
    className="c2"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    <span>
      <i
        className="far fa-trash-alt"
      />
    </span>
    <span>
      Delete
    </span>
  </button>
  <mock-ConfirmationModal
    isConfirming={false}
    isOpen={false}
    onCancel={[Function]}
    onConfirm={[Function]}
  >
    Are you sure you want to remove 
    <b>
      my-connector
    </b>
     connector?
  </mock-ConfirmationModal>
</div>
`;
